src/lib/crawler/crawler.integration.test.ts

```diff
@@ -0,0 +1,55 @@
+import { WarmaneCrawler } from "./crawler";
+
+describe("crawler integration tests", () => {
+  const character = "Dumpster";
+  const realm = "Blackrock";
+
+  // increasing timeout to allow requests to complete
+  jest.setTimeout(30000);
+
+  test("The crawler can return a list of match summaries with requests", async () => {
+    const crawler = new WarmaneCrawler();
+    const matchSummaries = await crawler.getMatchSummaries({
+      character,
+      realm,
+    });
+
+    expect(matchSummaries.length).toBeGreaterThan(0);
+    expect(matchSummaries[0]).toHaveProperty("matchId");
+  });
+
+  test("The crawler can return match details with requests", async () => {
+    const crawler = new WarmaneCrawler();
+    const matchSummaries = await crawler.getMatchSummaries({
+      character,
+      realm,
+    });
+
+    const matchDetailsList = await crawler.getMatchDetails({
+      character,
+      realm,
+      matchSummaries,
+    });
+
+    expect(matchDetailsList.length).toBeGreaterThan(0);
+    expect(matchDetailsList[0]).toHaveProperty("character_details");
+    expect(matchDetailsList[0].character_details.length).toBeGreaterThan(0);
+    expect(matchDetailsList[0].character_details[0]).toHaveProperty("charname");
+    expect(matchDetailsList[0].character_details[0]).toHaveProperty("realm");
```

Comment:
I actually like the idea of integration tests, but there are a few things we should consider here:

* Time it takes to run tests?
* Where do we run them inside of GitHub actions and how often?
* How do we prevent them from breaking over time?
* Will this potentially blow our cover or DDOS warmane?
* What does this integration test cover that our unit tests do not?

For the first one, I see you have already upped timeout to 30 seconds. If we're actually running the crawler that is fine, but know that some people play so many games that it might take more than 30 seconds. Also, we'll need to modify automation scripts or tell tooling not to run these tests unless we tell them to. Have you downloaded the Jest extension yet? If I recall, it tries to automatically run all tests whenever you save **any file**.

I think this is probably safe to run in GitHub actions, but it might be nice to start separating the integration with unit tests. For instance, the unit tests run ideally in less than one second assuming the TypeScript has already compiled. Unit tests should comprise a majority of our tests, so why should we wait for the 1 integration test to fail after 30 seconds when we should know almost immediately that one of the hundreds of unit tests is going to fail.

I think one of the biggest problems here is the assumption that Dumpster will always exist on Blackrock and will always have played at least one game. That is one of my characters and you're basically assuming that I play on that server a shit ton. Well you're not that far off :(

Still though, I would prefer tests to always pass. I want to spend as little time as possible maintaining things. When we complete this project in a few weeks, I want to be done with it. If I want to change something small years from now, I don't want to sit there debugging integration tests to get my build to pass again.

Finally I think that the only thing that this test should really tell us that our unit tests don't, is if Warmane changed their data structures or HTML output - which is very likely. BUT, you know what can tell us this much faster? Monitoring and alerting that we can set up in AWS. If the crawler suddenly can't find data, we should be throwing errors right? We can configure AWS to instantly alert us via text message, alexa skill, phone call, email, package delivery, or whatever at the instant errors exceed some threshold. I mean if the integration tests break, we know that the whole website is broken at the same time.

I have this blog post saved in my notes from a long time ago, but I still think it's relevant to this conversation. https://testing.googleblog.com/2015/04/just-say-no-to-more-end-to-end-tests.html

All of this being said, I'm fine with keeping these tests in for now.